### PR TITLE
fix(component): make passport packages optional

### LINKED ISSUE #125: All passport packages should be optional, and the user should install them only if required.
**Describe the bug**
Currently all the passport packages are installed for all the users, this should not happen ideally.

**To Reproduce**
1. install loopback4-authentication
2. Check the node-modules, it has all the passport strategies

**Expected behavior**
Only the packages required by the user should be installed

### PR DESCRIPTION
**Pull request by RaghavaroraSF**
## Description

- All passport packages are made optional, and the user should install them only if required.

BREAKING CHANGE:
 - change import path for specific passport strategies

GH-125

Fixes #125

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

- The following link shows how this has been tested :-
https://github.com/RaghavaroraSF/Cpassword

## Checklist:

- [x] Performed a self-review of my own code
- [x] npm test passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by shubhamp-sf** [comment]
Hi @RaghavaroraSF please include the repo link in the description of the repo in which you tested this change.

**Comment by yeshamavani** [comment]
@karuneshsf picked this issue

File: README.md [hunk 1]
```diff
@@ -1008,6 +1012,7 @@ For accessing the authenticated AuthUser model reference, you can inject the CUR
 
 ### Google Oauth 2
 
+In order to use it, run `npm install passport-google-oauth20` or `npm install @types/passport-google-oauth20`.
```
[the remark this answers is not in the document]
Comment (yeshamavani) [reply README.md:1025]: @karuneshsf

**Review by yeshamavani — CHANGES_REQUESTED** [review]

**Comment by yeshamavani** [comment]
code smells

**Review by yeshamavani — APPROVED** [review]

**Pull request merged**